### ISSUE TITLE
Only allow the `SynapsKYCDialog` on Metacolony

### ISSUE THREAD

**Issue by rdig** [JoinColony/colonyDapp#2947]
The current implementation for this just checks if the Metacolony is deployed, not if the current colony _is_ the Metacolony:

https://github.com/JoinColony/colonyDapp/blob/62cfc76779a52c9f86b5a07eac451b186e3a8937/src/modules/dashboard/components/CoinMachine/GetWhitelisted/GetWhitelisted.tsx#L71-L75

The `useMetaColonyQuery` returns the full Metacolony profile, if it's available on the chain _(eg: deployed)_

So this needs to change to check if the current colony that the user is on, is also the metacolony. 

Something like this:
```js
 const openKYCDialog = useCallback(() => { 
   return data?.processedMetaColony?.colonyAddress === colonyAddress
     ? openSynapsDialog({ colonyAddress }) 
     : openCompleteKYCDialog(); 
 }, [data, openSynapsDialog, colonyAddress, openCompleteKYCDialog]); 
```

**Comment by rdig** [comment]
What's the status of this, are you working on it @agustif ?

**Comment by agustif** [comment]
@rdig yes, will be pushing the suggested changes shortly and creating a new draft PR for it, but I haven't been able to test them properly yet. I will do so later today.
